src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -519,16 +519,15 @@ public FSDataInputStream getObject(String hostName, Path path) throws IOExceptio
         }
         String unifiedObjectName = extractUnifiedObjectName(tmp.getName());
         // was obj + "/"
-        if (!prefixBased && !obj.equals("") && !path.toString().endsWith("/")
+        if (!prefixBased && !obj.equals("") && path.toString().endsWith("/")
             && !unifiedObjectName.equals(obj) && !unifiedObjectName.startsWith(obj)) {
           // JOSS returns all objects that start with the prefix of obj.
           // These may include other unrelated objects.
           LOG.debug("{} does not match {}. Skipped", unifiedObjectName, obj);
           continue;
         }
         LOG.debug("Unified name: {}, path {}", unifiedObjectName, tmp.getName());
-        if (!unifiedObjectName.equals(tmp.getName()) && isSparkOrigin(unifiedObjectName)
-            && !fullListing) {
```

Comment:
@gilv I think I originally removed it because one of the tests failed, since it didn't upload the spark origin metadata. I rebased with upstream/hadoop-support and it works fine now. I'll update the branch.